### DISCUSSION TITLE
How to get octoprint to autostart after a reset or reboot

### DISCUSSION THREAD

**Discussion by Alf-Tupper** [paukstelis/octoprint_deploy#177] [Q&A] [answered]
Hi,

I have got octoprint installed and all seem to be working great, apart from I have to manually restart in terminal with "sudo systemctl restart" after I reset octoprint or reboot/shutdown system.

Can someone please tell me how to get this working.

Cheers

**Comment by paukstelis** [comment]
It is set up to automatically start when you install an instance. If you restored from a backup, it likely changed the command in OctoPrint.
It should always come up on a reboot, but you can make sure with: `sudo systemctl enable instancename`

**Comment by Alf-Tupper** [comment]
Thanks for the reply @paukstelis 

I have tried `sudo systemctl enable instancename` but it made no difference. The only way I can get octoprint to start up is with `sudo systemctl restart instancename`

Is there anything else you could suggest?

**Reply by paukstelis** [reply]
I would reboot and then see why it hasn't started. You can do a `sudo systemctl status instancename`.

**Reply by Alf-Tupper** [reply]
`$ sudo systemctl status ender3-1
ender3-1.service - The snappy web interface for your 3D printer
    Loaded: loaded (/etc/systemd/system/ender3-1.service, enabled)
    Active: inactive (dead)`

**Reply by paukstelis** [reply]
That suggests to me that it isn't actually enabled to start.
`sudo systemctl enable ender3-1.service`
Is there any output from that?

**Reply by Alf-Tupper** [reply]
No output, and I get the same status. If I run `sudo systemctl restart ender3-1`, octoprint starts and I get the status `$ sudo systemctl status ender3-1 ender3-1.service - The snappy web interface for your 3D printer
    Loaded: loaded (/etc/systemd/system/ender3-1.service, enabled)
    Active: active (running)`

**Comment by paukstelis** [comment]
What type of system, and base OS is this running on?

**Reply by Alf-Tupper** [reply]
The system is a dell studio hybrid 140g and the OS is antiX Linux

**Reply by paukstelis** [reply]
I think the problem is that antiX doesn't use systemd (by default, anyway).

**Reply by Alf-Tupper** [reply]
Ahh ok, will have a look into that. Thanks.

**Comment by Alf-Tupper** [comment, marked as answer]
Solved (for me at least). For anyone that may have this issue in the future, I followed the post below and edited the command in the file to `systemctl restart instancename` 

[https://askubuntu.com/questions/956237/run-terminal-sudo-command-at-startup](url)
